docker-entrypoint.sh

```diff
@@ -1,110 +1,129 @@
 echo "=="
-if [ -f "/var/www/localhost/htdocs/protected/config/dynamic.php" ]; then
-	echo "Existing installation found!"
-
-	wait_for_db
-
-	INSTALL_VERSION=$(cat /var/www/localhost/htdocs/protected/config/.version)
-	SOURCE_VERSION=$(cat /usr/src/humhub/.version)
-	cd /var/www/localhost/htdocs/protected/
-	if [[ $INSTALL_VERSION != $SOURCE_VERSION ]]; then
-		echo "Updating from version $INSTALL_VERSION to $SOURCE_VERSION"
-		php yii migrate/up --includeModuleMigrations=1 --interactive=0
-		php yii search/rebuild
-		cp -v /usr/src/humhub/.version /var/www/localhost/htdocs/protected/config/.version
-	fi
+# Look for root index.php rather than dynamic.php, which could be persisted with config files only
+if [ -f "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/index.php" ]; then
+  echo "Existing installation found!"
+
+  wait_for_db
+
+  INSTALL_VERSION="$(cat ${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/.version)"
+  SOURCE_VERSION="$(cat /tmp/humhub/.version)"
+  cd "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/protected"
+  if [[ $INSTALL_VERSION != $SOURCE_VERSION ]]; then
+    echo "Updating from version $INSTALL_VERSION to $SOURCE_VERSION"
+    php yii migrate/up --includeModuleMigrations=1 --interactive=0
+    php yii search/rebuild
+    cp -va /tmp/humhub/.version "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/.version"
+  fi
 else
-	echo "No existing installation found!"
-	echo "Installing source files..."
-	cp -rv /usr/src/humhub/protected/config/* /var/www/localhost/htdocs/protected/config/
-	cp -v /usr/src/humhub/.version /var/www/localhost/htdocs/protected/config/.version
-
-	mkdir -p /var/www/localhost/htdocs/protected/runtime/logs/
-	touch /var/www/localhost/htdocs/protected/runtime/logs/app.log
-
-	echo "Setting permissions..."
-	chown -R nginx:nginx /var/www/localhost/htdocs/uploads
-	chown -R nginx:nginx /var/www/localhost/htdocs/protected/modules
-	chown -R nginx:nginx /var/www/localhost/htdocs/protected/config
-	chown -R nginx:nginx /var/www/localhost/htdocs/protected/runtime
-
-	wait_for_db
-
-	echo "Creating database..."
-	cd /var/www/localhost/htdocs/protected/
-	if [ -z "$HUMHUB_DB_USER" ]; then
-		AUTOINSTALL="false"
-	fi
-
-	if [ "$AUTOINSTALL" != "false" ]; then
-		echo "Installing..."
-		php yii installer/write-db-config "$HUMHUB_DB_HOST" "$HUMHUB_DB_NAME" "$HUMHUB_DB_USER" "$HUMHUB_DB_PASSWORD"
-		php yii installer/install-db
-		php yii installer/write-site-config "$HUMHUB_NAME" "$HUMHUB_EMAIL"
-		php yii installer/create-admin-account
-		chown -R nginx:nginx /var/www/localhost/htdocs/protected/runtime
-	fi
+  echo "No existing installation found!"
+  echo "Installing source files..."
+  if [ ! -d "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}" ]; then
+    echo "Moving humhub files to ${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+    mv /tmp/humhub "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+  else
+    mkdir -p "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+    echo "Copying files into ${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}"
+    cp -a /tmp/humhub/* "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/"
```

Comment:
This counteracts the "new" auto-update mechanism. The whole idea is to avoid having to copy all files from some place inside the container to another place (maybe a volume). If all "static" files (everything except uploads, plugins and config) is stored inside the image, updating the container is fast and easy because it does not involve transferring any files but only doing some migrations and cache-clearing. Your code uses "the old way" and is therefor wasting some space for the /tmp/humhub files.

Reply:
Hm... I must not have understood how you were doing the auto-update mechanism. I'll take another look.